core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -311,7 +311,19 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
       getV2WorkspaceContextAndPermissions(workspaceName, SamWorkspaceActions.read, Option(attrSpecs)) flatMap {
         workspaceContext =>
           dataSource.inTransaction { dataAccess =>
-            val wsmContext = wsmService.optimizedFetchAggregatedWorkspace(workspaceContext, ctx)
+            // some GCP workspaces, like those with linked snapshots, have a stub WSM workspace
```

Comment:
Perhaps this is oversimplified now... I was trying to get rid of the "optimized" function and repurpose that logic here so that it'd check `fetchAggregatedWorkspace` for the rawls workspace too, but `fetchAggregatedWorkspace` could also fail on an MC workspace. Is there a way to modify the `catch` to be `if AggregateWorkspaceNotFoundException && WorkspaceType.RawlsWorkspace`?

Reply:
It looks like there's something similar in SnapshotService that might be helpful:

https://github.com/broadinstitute/rawls/blob/19c84937b0d48342577f51b4f229b4be12a02e48/core/src/main/scala/org/broadinstitute/dsde/rawls/snapshot/SnapshotService.scala#L122-L143

Reply:
ya know, I linked that exact line of code in the PR description and yet I missed how it was checking. thank you

Reply:
Oops, sorry, I missed that in the description.